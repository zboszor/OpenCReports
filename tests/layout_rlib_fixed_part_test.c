/*
 * OpenCReports test
 *
 * This is a copy of the "fixed_part" example from RLIB
 * ported to C code. The XML part is identical.
 *
 * Copyright (C) 2019-2022 Zoltán Böszörményi <zboszor@gmail.com>
 * See COPYING.LGPLv3 in the toplevel directory.
 */

#include <stdio.h>
#include <stdlib.h>

#include <opencreport.h>
#include "test_common.h"

char *yields[4][3] = {
	{ "item", "portions", "eqv" },
	{ "Hamburger", "1", "100" },
	{ "Chicken", "1", "150" },
	{ "Pasta", "1", "190" }
};

char *coupons[5][5] = {
	{ "name", "actual_count", "actual_amount", "computed_count", "computed_amount" },
	{ "DOLLAR OFF", "0", "0", "0", "0" },
	{ "FREE", "0", "0", "3", "-10.23" },
	{ "SENIOR 10%", "0", "0", "0", "0" },
	{ "15%", "0", "0", "0", "0" }
};

char *deposits[3][4] = {
	{ "time", "bag_id", "manager_id", "amount" },
	{ "12:00", "DEPOSIT 1", "101", "2000.00" },
	{ "4:00", "DEPOSIT 2", "102", "1000.00" }
};

char *petty_cash[3][3] = {
	{ "time", "name", "amount" },
	{ "1:00", "Sugar", "20.00" },
	{ "2:00", "Pants", "40.00" }
};

char *misc_income[2][3] = {
	{ "time", "name", "amount" },
	{ "1:00", "Birthday Party", "20.00" }
};

char *inv_transfer[3][4] = {
	{ "qty", "name", "from", "to" },
	{ "100", "Buns", "1121", "4452" },
	{ "400", "Pattys", "1121", "4499" }
};

char *inventory[3][11] = {
	{ "num", "name", "amount", "unit", "open", "usage", "received", "transfer_in", "transfer_out", "waste", "pysical_count" },
	{ "1", "BUN, REG/PREMIUM", ".50", "DOZEN", "176", "0", "0", "0", "0", "0", "0" },
	{ "2", "BUN, KAISER", ".70", "DOZEN", "176", "0", "0", "0", "0", "0", "0" }
};

void part_iter_cb(opencreport *o, ocrpt_part *p, void *data) {
	yields[3][0] = "MIKEROTH";
	yields[3][1] = "CALL";
	yields[3][2] = "BACK";
}

void precalc_done_cb(opencreport *o, void *data) {
	yields[3][0] = "Pasta";
	yields[3][1] = "1";
	yields[3][2] = "190";
}

int main(void) {
	opencreport *o = ocrpt_init();
	ocrpt_datasource *ds = ocrpt_datasource_add_array(o, "local_array");

	/* For m.sillypants in the report */
	setenv("sillypants", "5", 1);

	ocrpt_query_add_array(o, ds, "yields", (const char **)yields, 3, 3, NULL);
	ocrpt_query_add_array(o, ds, "coupons", (const char **)coupons, 4, 5, NULL);
	ocrpt_query_add_array(o, ds, "deposits", (const char **)deposits, 2, 4, NULL);
	ocrpt_query_add_array(o, ds, "petty_cash", (const char **)petty_cash, 2, 3, NULL);
	ocrpt_query_add_array(o, ds, "misc_income", (const char **)misc_income, 1, 3, NULL);
	ocrpt_query_add_array(o, ds, "inv_transfer", (const char **)inv_transfer, 2, 4, NULL);
	ocrpt_query_add_array(o, ds, "inventory", (const char **)inventory, 2, 11, NULL);

	if (!ocrpt_parse_xml(o, "layout_rlib_fixed_part_test.xml")) {
		printf("XML parse error\n");
		ocrpt_free(o);
		return 0;
	}

	ocrpt_add_part_iteration_cb(o, part_iter_cb, NULL);
	ocrpt_add_precalculation_done_cb(o, precalc_done_cb, NULL);

	ocrpt_set_output_format(o, OCRPT_OUTPUT_PDF);

	ocrpt_execute(o);

	ocrpt_spool(o);

	ocrpt_free(o);

	return 0;
}
