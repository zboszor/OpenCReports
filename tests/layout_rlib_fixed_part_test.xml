<?xml version="1.0"?>
<!DOCTYPE report>
<Part layout="'fixed'" fontSize="14" orientation="landscape" iterations="2">
  <PageHeader>
    <Output>
      <Line fontSize="26">
        <literal>Fixed Part Test</literal>
      </Line>
      <Line>
        <literal>ICR</literal>
      </Line>
      <HorizontalLine size="4" bgcolor="'black'"/>
      <HorizontalLine size="10" bgcolor="'white'"/>
    </Output>
  </PageHeader>
  <pr>
    <pd>
      <Report fontSize="9" query="'inventory'">
        <Alternate>
          <NoData>
            <Output>
              <Line fontSize="12">
                <literal>NO DATA</literal>
              </Line>
            </Output>
          </NoData>
        </Alternate>
        <ReportHeader>
          <Output>
            <Line/>
            <Line fontSize="9">
              <literal>      INVENTORY REPORT</literal>
            </Line>
            <HorizontalLine size="2" bgcolor="'white'"/>
          </Output>
        </ReportHeader>
        <Detail>
          <FieldHeaders>
            <Output>
              <HorizontalLine size="1" bgcolor="'black'"/>
              <Line bgcolor="'0xe5e5e5'">
                <literal width="3" col="1">NUM</literal>
                <literal width="1"/>
                <literal width="20" col="2">NAME</literal>
                <literal width="1"/>
                <literal width="10" col="3">UNIT</literal>
                <literal width="1"/>
                <literal width="10" col="4">OPEN</literal>
                <literal width="1"/>
                <literal width="10" col="5">USAGE</literal>
                <literal width="1"/>
                <literal width="10" col="6">RECEIVED</literal>
                <literal width="1"/>
                <literal width="10" col="7">TRANS IN</literal>
                <literal width="1"/>
                <literal width="10" col="8">TRANS OUT</literal>
                <literal width="1"/>
                <literal width="10" col="9">WASTE</literal>
                <literal width="1"/>
              </Line>
              <HorizontalLine size="1" bgcolor="'black'"/>
              <HorizontalLine size="4" bgcolor="'white'"/>
            </Output>
          </FieldHeaders>
          <FieldDetails>
            <Output>
              <Line bgcolor="iif(r.detailcnt%2,'0xe5e5e5','white')">
                <field value="num" width="3" align="left" col="1"/>
                <literal width="1"/>
                <field value="name" width="20" align="left" col="2"/>
                <literal width="1"/>
                <field value="unit" width="10" align="left" col="3"/>
                <literal width="1"/>
                <field value="open" width="10" align="left" col="4"/>
                <literal width="1"/>
                <field value="usage" width="10" align="left" col="5"/>
                <literal width="1"/>
                <field value="received" width="10" align="left" col="6"/>
                <literal width="1"/>
                <field value="transfer_in" width="10" align="left" col="7"/>
                <literal width="1"/>
                <field value="transfer_out" width="10" align="left" col="8"/>
                <literal width="1"/>
                <field value="waste" width="10" align="left" col="9"/>
                <literal width="1"/>
              </Line>
            </Output>
          </FieldDetails>
        </Detail>
      </Report>
    </pd>
  </pr>
  <pr newpage="yes">
    <pd width="30" height="81" border_width="2" border_color="'black'">
      <Report fontSize="9" query="'yields'" height="19" leftMargin="0.1">
        <Alternate>
          <NoData>
            <Output>
              <Line fontSize="12">
                <literal>NO DATA</literal>
              </Line>
            </Output>
          </NoData>
        </Alternate>
        <ReportHeader>
          <Output>
            <Line fontSize="9">
              <literal>YIELDS</literal>
            </Line>
            <HorizontalLine size="2" bgcolor="'white'"/>
          </Output>
        </ReportHeader>
        <Detail>
          <FieldHeaders>
            <Output>
              <HorizontalLine size="1" bgcolor="'black'" length="39"/>
              <Line bgcolor="'0xe5e5e5'">
                <literal width="10" col="1">Item</literal>
                <literal width="1"/>
                <literal width="9" col="2">Portions</literal>
                <literal width="1"/>
                <literal width="7" col="3">EQV</literal>
                <literal width="1"/>
                <literal width="10" col="4">REG EQUIV</literal>
              </Line>
              <HorizontalLine size="1" bgcolor="'black'" length="39"/>
              <HorizontalLine size="4" bgcolor="'white'"/>
            </Output>
          </FieldHeaders>
          <FieldDetails>
            <Output>
              <Line bgcolor="iif(r.detailcnt%2,'0xe5e5e5','white')">
                <field value="item" width="10" align="left" col="1"/>
                <literal width="1"/>
                <field value="portions" width="9" align="left" col="2"/>
                <literal width="1"/>
                <field value="eqv" width="10" align="left" col="3"/>
              </Line>
            </Output>
          </FieldDetails>
        </Detail>
        <ReportFooter>
          <Output>
            <Line fontSize="9">
              <literal>      YIELDS</literal>
            </Line>
            <Line fontSize="9">
              <literal>      YIELDS</literal>
            </Line>
            <Line fontSize="9">
              <literal>      YIELDS</literal>
            </Line>
          </Output>
        </ReportFooter>
      </Report>
      <Report fontSize="9" query="'coupons'" height="32" leftMargin="0.1" suppress="yes">
        <Alternate>
          <NoData>
            <Output>
              <Line fontSize="12">
                <literal>NO DATA</literal>
              </Line>
            </Output>
          </NoData>
        </Alternate>
        <ReportHeader>
          <Output>
            <Line/>
            <Line fontSize="9">
              <literal>      COUPONSXX</literal>
            </Line>
            <HorizontalLine size="2" bgcolor="'white'"/>
          </Output>
        </ReportHeader>
        <Detail>
          <FieldHeaders>
            <Output>
              <HorizontalLine size="1" bgcolor="'black'" length="39"/>
              <Line bgcolor="'0xe5e5e5'">
                <literal width="15" col="1">Name</literal>
                <literal width="1"/>
                <literal width="11" col="2">   PHY.CNT </literal>
                <literal width="1"/>
                <literal width="11" col="3">      POS    </literal>
              </Line>
              <HorizontalLine size="1" bgcolor="'black'" length="39"/>
              <HorizontalLine size="4" bgcolor="'white'"/>
            </Output>
          </FieldHeaders>
          <FieldDetails>
            <Output>
              <Line bgcolor="iif(r.detailcnt%2,'0xe5e5e5','white')">
                <field value="name" width="15" align="left" col="1"/>
                <literal width="1"/>
                <field value="actual_amount" width="11" align="left" col="2"/>
                <literal width="1"/>
                <field value="computed_amount" width="11" align="left" col="3"/>
              </Line>
            </Output>
          </FieldDetails>
        </Detail>
      </Report>
      <Report fontSize="9" query="'dttiming'" height="20" leftMargin="0.1">
        <Alternate>
          <NoData>
            <Output>
              <Line fontSize="12">
                <literal>NO DATA</literal>
              </Line>
            </Output>
          </NoData>
        </Alternate>
        <ReportHeader>
          <Output>
            <Line/>
            <Line>
              <literal>   WWW.DT TIMING</literal>
              <field value="m.sillypants" width="5"/>
            </Line>
            <Line fontSize="9">
              <literal>NUMBER OF CARS</literal>
            </Line>
            <Line fontSize="9">
              <literal>NUMBER PARKED</literal>
            </Line>
            <Line/>
            <Line fontSize="9">
              <literal>EXCEED COUNT</literal>
            </Line>
            <Line fontSize="9">
              <literal>NUMBER PARKED</literal>
            </Line>
            <Line/>
            <Line fontSize="9">
              <literal>AVE PARK TIME</literal>
            </Line>
            <Line fontSize="9">
              <literal>AVE TOTAL TIME</literal>
            </Line>
            <HorizontalLine size="2" bgcolor="'white'"/>
          </Output>
        </ReportHeader>
      </Report>
    </pd>
    <pd width="30" height="81" border_width="2" border_color="'black'">
      <Report fontSize="9" query="'deposits'" height="17" leftMargin="0.1">
        <Alternate>
          <NoData>
            <Output>
              <Line fontSize="12">
                <literal>NO DATA</literal>
              </Line>
            </Output>
          </NoData>
        </Alternate>
        <ReportHeader>
          <Output>
            <Line fontSize="9">
              <literal>DEPOSITS</literal>
            </Line>
            <HorizontalLine size="2" bgcolor="'white'"/>
          </Output>
        </ReportHeader>
        <Detail>
          <FieldHeaders>
            <Output>
              <HorizontalLine size="1" bgcolor="'black'" length="39"/>
              <Line bgcolor="'0xe5e5e5'">
                <literal width="8" col="1">  TIME  </literal>
                <literal width="1"/>
                <literal width="10" col="2"> BAG ID   </literal>
                <literal width="1"/>
                <literal width="9" col="3"> MGR. ID </literal>
                <literal width="1"/>
                <literal width="10" col="4">   AMOUNT</literal>
              </Line>
              <HorizontalLine size="1" bgcolor="'black'" length="39"/>
              <HorizontalLine size="4" bgcolor="'white'"/>
            </Output>
          </FieldHeaders>
          <FieldDetails>
            <Output>
              <Line bgcolor="iif(r.detailcnt%2,'0xe5e5e5','white')">
                <field value="time" width="8" align="left" col="1"/>
                <literal width="1"/>
                <field value="bag_id" width="10" align="left" col="2"/>
                <literal width="1"/>
                <field value="manager_id" width="9" align="left" col="3"/>
                <literal width="1"/>
                <field value="amount" width="10" align="left" col="4"/>
              </Line>
            </Output>
          </FieldDetails>
        </Detail>
        <ReportFooter>
          <Output>
            <Line fontSize="9">
              <literal> TOTAL DEPOSITS</literal>
            </Line>
          </Output>
        </ReportFooter>
      </Report>
      <Report fontSize="9" query="'other'" height="41" leftMargin="0.1">
        <Alternate>
          <NoData>
            <Output>
              <Line fontSize="12">
                <literal>NO DATA</literal>
              </Line>
            </Output>
          </NoData>
        </Alternate>
        <ReportHeader>
          <Output>
            <Line/>
            <Line>
              <literal>OTHER??? </literal>
            </Line>
            <Line fontSize="9">
              <literal>BREAK FAST SALES</literal>
            </Line>
            <Line fontSize="9">
              <literal>BREAKFAST SALES</literal>
            </Line>
            <Line fontSize="9">
              <literal>SALAD SALES</literal>
            </Line>
            <Line fontSize="9">
              <literal>SALAD SALES</literal>
            </Line>
            <Line fontSize="9">
              <literal>DRIVE-THRU SALES</literal>
            </Line>
            <Line fontSize="9">
              <literal>DRIVE-THRU SALES</literal>
            </Line>
            <Line fontSize="9">
              <literal>OVER-RINGS</literal>
            </Line>
            <Line fontSize="9">
              <literal>OVER-RINGS</literal>
            </Line>
            <Line fontSize="9">
              <literal>OVER-RINGS</literal>
            </Line>
            <Line fontSize="9">
              <literal>COUPONS MEALS WASTE</literal>
            </Line>
            <Line fontSize="9">
              <literal>COUPONS MEALS WASTE</literal>
            </Line>
            <HorizontalLine size="2" bgcolor="'white'"/>
          </Output>
        </ReportHeader>
      </Report>
      <Report fontSize="9" query="'inv_transfer'" height="15" leftMargin="0.1">
        <Alternate>
          <NoData>
            <Output>
              <Line fontSize="12">
                <literal>NO DATA</literal>
              </Line>
            </Output>
          </NoData>
        </Alternate>
        <ReportHeader>
          <Output>
            <Line/>
            <Line fontSize="9">
              <literal>      INVENTORY TRANSFER</literal>
            </Line>
            <HorizontalLine size="2" bgcolor="'white'"/>
          </Output>
        </ReportHeader>
        <Detail>
          <FieldHeaders>
            <Output>
              <HorizontalLine size="1" bgcolor="'black'" length="31"/>
              <Line bgcolor="'0xe5e5e5'">
                <literal width="5" col="1">QTY</literal>
                <literal width="1"/>
                <literal width="15" col="2">Name</literal>
                <literal width="1"/>
                <literal width="4" col="3">FROM</literal>
                <literal width="1"/>
                <literal width="4" col="4">TO</literal>
              </Line>
              <HorizontalLine size="1" bgcolor="'black'" length="31"/>
              <HorizontalLine size="4" bgcolor="'white'"/>
            </Output>
          </FieldHeaders>
          <FieldDetails>
            <Output>
              <Line bgcolor="iif(r.detailcnt%2,'0xe5e5e5','white')">
                <field value="qty" width="5" align="left" col="1"/>
                <literal width="1"/>
                <field value="name" width="15" align="left" col="2"/>
                <literal width="1"/>
                <field value="from" width="4" align="left" col="3"/>
                <literal width="1"/>
                <field value="to" width="4" align="left" col="4"/>
              </Line>
            </Output>
          </FieldDetails>
        </Detail>
      </Report>
    </pd>
    <pd width="40" height="81" border_width="2" border_color="'black'">
      <Report fontSize="9" query="'cash'" height="47" leftMargin="0.1">
        <Alternate>
          <NoData>
            <Output>
              <Line fontSize="12">
                <literal>NO DATA</literal>
              </Line>
            </Output>
          </NoData>
        </Alternate>
        <ReportHeader>
          <Output>
            <Line>
              <literal>     CASH RECONCILIATION </literal>
            </Line>
            <Line fontSize="9">
              <literal>NET SALES </literal>
            </Line>
            <Line fontSize="9">
              <literal>TAX 	</literal>
            </Line>
            <Line fontSize="9">
              <literal>GROSS SALES </literal>
            </Line>
            <Line fontSize="9">
              <literal>RESTAURANT BANK</literal>
            </Line>
            <Line fontSize="9">
              <literal>ADJUSTED GROSS CASH</literal>
            </Line>
            <Line fontSize="9">
              <literal>AMOUNT DEPOSITED</literal>
            </Line>
            <Line fontSize="9">
              <literal>GIFT REDEEMED</literal>
            </Line>
            <Line fontSize="9">
              <literal>MISC. INCOME</literal>
            </Line>
            <Line fontSize="9">
              <literal>PETTY CASH</literal>
            </Line>
            <Line fontSize="9">
              <literal>NET SALES</literal>
            </Line>
            <Line fontSize="9">
              <literal>TAX</literal>
            </Line>
            <Line fontSize="9">
              <literal>CASH OVER/SHORT </literal>
            </Line>
            <Line fontSize="9">
              <literal>CASHIER OVER/SHORT </literal>
            </Line>
            <Line fontSize="9">
              <literal>OVER/SHORT RECONCILIATION</literal>
            </Line>
            <Line fontSize="9">
              <literal>NET SALES </literal>
            </Line>
            <Line fontSize="9">
              <literal>GIFT CERT. SOLD </literal>
            </Line>
            <Line fontSize="9">
              <literal>OTHER NON-FOOD</literal>
            </Line>
            <Line fontSize="9">
              <literal>NET FOOD SALES</literal>
            </Line>
            <HorizontalLine size="2" bgcolor="'white'"/>
          </Output>
        </ReportHeader>
      </Report>
      <Report fontSize="9" query="'petty_cash'" leftMargin="0.1">
        <Alternate>
          <NoData>
            <Output>
              <Line fontSize="12">
                <literal>NO DATA</literal>
              </Line>
            </Output>
          </NoData>
        </Alternate>
        <ReportHeader>
          <Output>
            <Line/>
            <Line fontSize="9">
              <literal>      PETTY CASH</literal>
            </Line>
            <HorizontalLine size="2" bgcolor="'white'"/>
          </Output>
        </ReportHeader>
        <Detail>
          <FieldHeaders>
            <Output>
              <HorizontalLine size="1" bgcolor="'black'" length="42"/>
              <Line bgcolor="'0xe5e5e5'">
                <literal width="8" col="1">  TIME  </literal>
                <literal width="1"/>
                <literal width="20" col="2"> NAME   </literal>
                <literal width="1"/>
                <literal width="12" col="3">    AMOUNT  </literal>
              </Line>
              <HorizontalLine size="1" bgcolor="'black'" length="42"/>
              <HorizontalLine size="4" bgcolor="'white'"/>
            </Output>
          </FieldHeaders>
          <FieldDetails>
            <Output>
              <Line bgcolor="iif(r.detailcnt%2,'0xe5e5e5','white')">
                <field value="time" width="8" align="left" col="1"/>
                <literal width="1"/>
                <field value="name" width="20" align="left" col="2"/>
                <literal width="1"/>
                <field value="amount" width="12" align="left" col="3"/>
              </Line>
            </Output>
          </FieldDetails>
        </Detail>
        <ReportFooter>
          <Output>
            <Line fontSize="9">
              <literal> TOTAL PETTY CASH</literal>
            </Line>
          </Output>
        </ReportFooter>
      </Report>
      <Report fontSize="9" query="'misc_income'" leftMargin="0.1">
        <Alternate>
          <NoData>
            <Output>
              <Line fontSize="12">
                <literal>NO DATA</literal>
              </Line>
            </Output>
          </NoData>
        </Alternate>
        <ReportHeader>
          <Output>
            <Line/>
            <Line fontSize="9">
              <literal>      MISC INCOME</literal>
            </Line>
            <HorizontalLine size="2" bgcolor="'white'"/>
          </Output>
        </ReportHeader>
        <Detail>
          <FieldHeaders>
            <Output>
              <HorizontalLine size="1" bgcolor="'black'" length="42"/>
              <Line bgcolor="'0xe5e5e5'">
                <literal width="8" col="1">  TIME  </literal>
                <literal width="1"/>
                <literal width="20" col="2"> NAME   </literal>
                <literal width="1"/>
                <literal width="12" col="3">    AMOUNT  </literal>
              </Line>
              <HorizontalLine size="1" bgcolor="'black'" length="42"/>
              <HorizontalLine size="4" bgcolor="'white'"/>
            </Output>
          </FieldHeaders>
          <FieldDetails>
            <Output>
              <Line bgcolor="iif(r.detailcnt%2,'0xe5e5e5','white')">
                <field value="time" width="8" align="left" col="1"/>
                <literal width="1"/>
                <field value="name" width="20" align="left" col="2"/>
                <literal width="1"/>
                <field value="amount" width="12" align="left" col="3"/>
              </Line>
            </Output>
          </FieldDetails>
        </Detail>
        <ReportFooter>
          <Output>
            <Line fontSize="9">
              <literal> TOTAL MISC INCOME</literal>
            </Line>
          </Output>
        </ReportFooter>
      </Report>
    </pd>
  </pr>
  <pr>
    <pd width="30" height="15" border_width="2" border_color="'black'">
      <Report fontSize="9" query="'labor'" leftMargin="0.1">
        <Alternate>
          <NoData>
            <Output>
              <Line fontSize="12">
                <literal>NO DATA</literal>
              </Line>
            </Output>
          </NoData>
        </Alternate>
        <ReportHeader>
          <Output>
            <Line/>
            <Line>
              <literal>     LABOR </literal>
            </Line>
            <Line fontSize="9">
              <literal>FORMULA HRS</literal>
            </Line>
            <Line fontSize="9">
              <literal>ACTUAL HRS</literal>
            </Line>
            <Line fontSize="9">
              <literal>DIFFERENCE</literal>
            </Line>
            <HorizontalLine size="2" bgcolor="'white'"/>
          </Output>
        </ReportHeader>
      </Report>
    </pd>
    <pd width="70" height="15" border_width="2" border_color="'black'">
      <Report fontSize="9" leftMargin="0.1">
        <Alternate>
          <NoData>
            <Output>
              <Line fontSize="12">
                <literal>NO DATA</literal>
              </Line>
            </Output>
          </NoData>
        </Alternate>
        <ReportHeader>
          <Output>
            <Line/>
            <Line>
              <literal> ---------    Comments  ------------------</literal>
            </Line>
            <Line fontSize="9">
              <literal>slashdot.org</literal>
            </Line>
            <Line fontSize="9">
              <literal>philadelphiaeagles.com</literal>
            </Line>
            <Line fontSize="9">
              <literal>sicom.com                  EAT               MORE                 CHEESE</literal>
            </Line>
            <HorizontalLine size="2" bgcolor="'white'"/>
          </Output>
        </ReportHeader>
      </Report>
    </pd>
  </pr>
  <pr newpage="yes">
    <pd>
      <Report fontSize="9" query="'inventory'">
        <Alternate>
          <NoData>
            <Output>
              <Line fontSize="12">
                <literal>NO DATA</literal>
              </Line>
            </Output>
          </NoData>
        </Alternate>
        <ReportHeader>
          <Output>
            <Line/>
            <Line fontSize="9">
              <literal>      INVENTORY REPORT</literal>
            </Line>
            <HorizontalLine size="2" bgcolor="'white'"/>
          </Output>
        </ReportHeader>
        <Detail>
          <FieldHeaders>
            <Output>
              <HorizontalLine size="1" bgcolor="'black'"/>
              <Line bgcolor="'0xe5e5e5'">
                <literal width="3" col="1">NUM</literal>
                <literal width="1"/>
                <literal width="20" col="2">NAME</literal>
                <literal width="1"/>
                <literal width="10" col="3">UNIT</literal>
                <literal width="1"/>
                <literal width="10" col="4">OPEN</literal>
                <literal width="1"/>
                <literal width="10" col="5">USAGE</literal>
                <literal width="1"/>
                <literal width="10" col="6">RECEIVED</literal>
                <literal width="1"/>
                <literal width="10" col="7">TRANS IN</literal>
                <literal width="1"/>
                <literal width="10" col="8">TRANS OUT</literal>
                <literal width="1"/>
                <literal width="10" col="9">WASTE</literal>
                <literal width="1"/>
              </Line>
              <HorizontalLine size="1" bgcolor="'black'"/>
              <HorizontalLine size="4" bgcolor="'white'"/>
            </Output>
          </FieldHeaders>
          <FieldDetails>
            <Output>
              <Line bgcolor="iif(r.detailcnt%2,'0xe5e5e5','white')">
                <field value="num" width="3" align="left" col="1"/>
                <literal width="1"/>
                <field value="name" width="20" align="left" col="2"/>
                <literal width="1"/>
                <field value="unit" width="10" align="left" col="3"/>
                <literal width="1"/>
                <field value="open" width="10" align="left" col="4"/>
                <literal width="1"/>
                <field value="usage" width="10" align="left" col="5"/>
                <literal width="1"/>
                <field value="received" width="10" align="left" col="6"/>
                <literal width="1"/>
                <field value="transfer_in" width="10" align="left" col="7"/>
                <literal width="1"/>
                <field value="transfer_out" width="10" align="left" col="8"/>
                <literal width="1"/>
                <field value="waste" width="10" align="left" col="9"/>
                <literal width="1"/>
              </Line>
            </Output>
          </FieldDetails>
        </Detail>
      </Report>
    </pd>
  </pr>
  <PageFooter>
    <Output>
      <Line>
        <literal>Page: </literal>
        <field value="r.pageno" width="3" align="right"/>
      </Line>
      <Line>
        <literal>Wheeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee: </literal>
        <field value="r.pageno" width="3" align="right"/>
      </Line>
    </Output>
  </PageFooter>
</Part>
