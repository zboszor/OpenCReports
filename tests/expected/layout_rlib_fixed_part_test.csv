"Fixed Part Test"
"ICR"
"      INVENTORY REPORT"
"NUM","NAME","UNIT","OPEN","USAGE","RECEIVED","TRANS IN","TRANS OUT","WASTE"
"1","BUN, REG/PREMIUM","DOZEN","176","0","0","0","0","0"
"2","BUN, KAISER","DOZEN","176","0","0","0","0","0"
"YIELDS"
"Item","Portions","EQV","REG EQUIV"
"Hamburger","1","100"
"Chicken","1","150"
"Pasta","1","190"
"      YIELDS"
"      YIELDS"
"      YIELDS"
"   WWW.DT TIMING","5"
"NUMBER OF CARS"
"NUMBER PARKED"
"EXCEED COUNT"
"NUMBER PARKED"
"AVE PARK TIME"
"AVE TOTAL TIME"
"DEPOSITS"
"  TIME  "," BAG ID   "," MGR. ID ","   AMOUNT"
"12:00","DEPOSIT 1","101","2000.00"
"4:00","DEPOSIT 2","102","1000.00"
" TOTAL DEPOSITS"
"OTHER??? "
"BREAK FAST SALES"
"BREAKFAST SALES"
"SALAD SALES"
"SALAD SALES"
"DRIVE-THRU SALES"
"DRIVE-THRU SALES"
"OVER-RINGS"
"OVER-RINGS"
"OVER-RINGS"
"COUPONS MEALS WASTE"
"COUPONS MEALS WASTE"
"      INVENTORY TRANSFER"
"QTY","Name","FROM","TO"
"100","Buns","1121","4452"
"400","Pattys","1121","4499"
"     CASH RECONCILIATION "
"NET SALES "
"TAX 	"
"GROSS SALES "
"RESTAURANT BANK"
"ADJUSTED GROSS CASH"
"AMOUNT DEPOSITED"
"GIFT REDEEMED"
"MISC. INCOME"
"PETTY CASH"
"NET SALES"
"TAX"
"CASH OVER/SHORT "
"CASHIER OVER/SHORT "
"OVER/SHORT RECONCILIATION"
"NET SALES "
"GIFT CERT. SOLD "
"OTHER NON-FOOD"
"NET FOOD SALES"
"      PETTY CASH"
"  TIME  "," NAME   ","    AMOUNT  "
"1:00","Sugar","20.00"
"2:00","Pants","40.00"
" TOTAL PETTY CASH"
"      MISC INCOME"
"  TIME  "," NAME   ","    AMOUNT  "
"1:00","Birthday Party","20.00"
" TOTAL MISC INCOME"
"     LABOR "
"FORMULA HRS"
"ACTUAL HRS"
"DIFFERENCE"
" ---------    Comments  ------------------"
"slashdot.org"
"philadelphiaeagles.com"
"sicom.com                  EAT               MORE                 CHEESE"
"      INVENTORY REPORT"
"NUM","NAME","UNIT","OPEN","USAGE","RECEIVED","TRANS IN","TRANS OUT","WASTE"
"1","BUN, REG/PREMIUM","DOZEN","176","0","0","0","0","0"
"2","BUN, KAISER","DOZEN","176","0","0","0","0","0"
"Page: ","1"
"Wheeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee: ","1"
"      INVENTORY REPORT"
"NUM","NAME","UNIT","OPEN","USAGE","RECEIVED","TRANS IN","TRANS OUT","WASTE"
"1","BUN, REG/PREMIUM","DOZEN","176","0","0","0","0","0"
"2","BUN, KAISER","DOZEN","176","0","0","0","0","0"
"YIELDS"
"Item","Portions","EQV","REG EQUIV"
"Hamburger","1","100"
"Chicken","1","150"
"MIKEROTH","CALL","BACK"
"      YIELDS"
"      YIELDS"
"      YIELDS"
"   WWW.DT TIMING","5"
"NUMBER OF CARS"
"NUMBER PARKED"
"EXCEED COUNT"
"NUMBER PARKED"
"AVE PARK TIME"
"AVE TOTAL TIME"
"DEPOSITS"
"  TIME  "," BAG ID   "," MGR. ID ","   AMOUNT"
"12:00","DEPOSIT 1","101","2000.00"
"4:00","DEPOSIT 2","102","1000.00"
" TOTAL DEPOSITS"
"OTHER??? "
"BREAK FAST SALES"
"BREAKFAST SALES"
"SALAD SALES"
"SALAD SALES"
"DRIVE-THRU SALES"
"DRIVE-THRU SALES"
"OVER-RINGS"
"OVER-RINGS"
"OVER-RINGS"
"COUPONS MEALS WASTE"
"COUPONS MEALS WASTE"
"      INVENTORY TRANSFER"
"QTY","Name","FROM","TO"
"100","Buns","1121","4452"
"400","Pattys","1121","4499"
"     CASH RECONCILIATION "
"NET SALES "
"TAX 	"
"GROSS SALES "
"RESTAURANT BANK"
"ADJUSTED GROSS CASH"
"AMOUNT DEPOSITED"
"GIFT REDEEMED"
"MISC. INCOME"
"PETTY CASH"
"NET SALES"
"TAX"
"CASH OVER/SHORT "
"CASHIER OVER/SHORT "
"OVER/SHORT RECONCILIATION"
"NET SALES "
"GIFT CERT. SOLD "
"OTHER NON-FOOD"
"NET FOOD SALES"
"      PETTY CASH"
"  TIME  "," NAME   ","    AMOUNT  "
"1:00","Sugar","20.00"
"2:00","Pants","40.00"
" TOTAL PETTY CASH"
"      MISC INCOME"
"  TIME  "," NAME   ","    AMOUNT  "
"1:00","Birthday Party","20.00"
" TOTAL MISC INCOME"
"     LABOR "
"FORMULA HRS"
"ACTUAL HRS"
"DIFFERENCE"
" ---------    Comments  ------------------"
"slashdot.org"
"philadelphiaeagles.com"
"sicom.com                  EAT               MORE                 CHEESE"
"      INVENTORY REPORT"
"NUM","NAME","UNIT","OPEN","USAGE","RECEIVED","TRANS IN","TRANS OUT","WASTE"
"1","BUN, REG/PREMIUM","DOZEN","176","0","0","0","0","0"
"2","BUN, KAISER","DOZEN","176","0","0","0","0","0"
"Page: ","1"
"Wheeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee: ","1"
